src/analysis/rust_type.rs

```diff
@@ -191,11 +196,59 @@ fn rust_type_full(
             }
         }
         Custom(library::Custom { ref name, .. }) => Ok(name.clone()),
+        Function(ref f) => {
+            let full_name = type_id.full_name(&env.library);
+            if full_name == "Gio.AsyncReadyCallback" {
+                return Ok(format!("FnOnce(Result<(), Error>) + Send + 'static"));
+            }
+            if full_name == "GLib.DestroyNotify" {
+                return Ok(format!("Fn() + 'static"));
+            }
+            let mut s = Vec::with_capacity(f.parameters.len());
+            let mut err = false;
+            for p in f.parameters.iter() {
+                if p.closure.is_some() {
+                    continue
+                }
+                match rust_type(env, p.typ) {
+                    Ok(x) => s.push(x),
+                    e => {
+                        err = true;
+                        s.push(e.into_string());
+                    }
+                }
+            }
+            let ret = match rust_type(env, f.ret.typ) {
+                Ok(x) => format!("{}({}) -> {}",
+                                 if scope.is_call() { "FnMut" } else { "Fn" },
+                                 s.join(", "),
+                                 if x != "GString" { &x } else { "String" }),
+                Err(TypeError::Unimplemented(ref x)) if x == "()" => {
+                    format!("{}({})",
+                            if scope.is_call() { "FnMut" } else { "Fn" },
+                            s.join(", "))
+                }
+                e => {
+                    err = true;
+                    format!("{}({}) -> {}",
+                            if scope.is_call() { "FnMut" } else { "Fn" },
+                            s.join(", "),
+                            e.into_string())
```

Comment:
Instead of formatting here, can't you just `return Err(...)` directly?

Reply:
It allows to know where the error comes from (by adding the comment) instead of just having the callback type wrote out and not additional information.